docs/fastdds/ros2/Discovery_service/ros2_discovery_service.rst

```diff
@@ -0,0 +1,337 @@
+.. ros2-discovery-service
+
+Use ROS2 with Fast-DDS Discovery Service
+========================================
+
+In this section we are going to explain how to run some ROS2 examples using the Discovery Services
+as discovery communication.
+In order to get more information about the specific use of this configuration,
+please check the :ref:`Discovery Service Documentation <discovery_service>`
+or read the :ref:`common use cases <discovery-service-use-case>` for this configuration.
+
+The following tutorial gathers the steps to check this functionality and learn how to use it with ROS2.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 2
+
+The :ref:`Simple Discovery Protocol <simple_disc_settings>` is the
+standard protocol defined in the `standard DDS <https://www.omg.org/omg-dds-portal/>`__.
+But it has certain known disadvantages in some scenarios:
+
+* It does not **Scale** efficiently, as the number of exchanged packets highly increases as new nodes are added.
+* It requires **Multicasting** capabilities, that may not work reliably in some connection cases as WiFi.
+
+The **Discovery Service** provides a Client-Server Architecture that allows
+the nodes to connect with each other using an intermediate server.
+Each node will work as a *Client*, sharing its info with the *Discovery Server* and receiving
+the discovery information from it.
+This means that the network traffic is highly reduced in big systems, and it does not require *Multicasting*.
+
+.. image:: ds_explanation.png
+    :align: center
+
+These **Discovery Servers** can be independent, duplicated or connected with each other in order to create
+redundancy over the network and avoid having a *Single-Point-Of-Failure*.
+
+In order to use this functionality, **Fast-DDS Discovery Service** can be set by parameters using
+the :ref:`XML configuration for Participants <DS_setup_concepts>`.
+
+However, Fast DDS library provides an easier way to set a **Discovery Service** communication using
+the ``fastdds`` :ref:`CLI tool <ffastddscli_cli>`
+and an :ref:`environment variable <env_vars>`.
+
+For a complete information about the actual implementation, testing and performance,
+visit this `link <https://eprosima.com/index.php/resources-all/scalability/fast-rtps-discovery-mechanisms-analysis>`__
+
+
+Prerequisites
+-------------
+
+This tutorial assumes you have a `working ROS 2 installation <https://index.ros.org/doc/ros2/Installation/>`__.
+In case your installation is using a Fast DDS version lower than v2.1.0, you could not use the ``fastdds`` tool.
```

Comment:
CLI is coming on v2.0.2 as well. Besides, Foxy from source already has this tool